### ISSUE TITLE
python2-urllib2 stub - ERROR self-signed certificate [reject expired.tlsfun.de:443]

### ISSUE THREAD

**Issue by ikisusi** [ouspg/trytls#229]
```
# trytls https python python2-urllib2/run.py
platform: Linux (CentOS Linux 7.2.1511)
runner: trytls 0.3.2 (CPython 2.7.5, OpenSSL 1.0.1e-fips)
stub: python python2-urllib2/run.py
 PASS protect against Apple's TLS vulnerability CVE-2014-1266 [reject www.ssllabs.com:10443]
 PASS protect against the FREAK attack [reject www.ssllabs.com:10444]
 PASS protect against the Logjam attack [reject www.ssllabs.com:10445]
 PASS protect against FREAK attack (test server 1) [reject cve.freakattack.com:443]
 PASS protect against FREAK attack (test server 2) [reject cve2.freakattack.com:443]
 PASS protection against POODLE attack [accept sslv3.dshield.org:443]
 PASS support for TLS server name indication (SNI) [accept badssl.com:443]
 FAIL self-signed certificate [reject self-signed.badssl.com:443]
 SKIP expired certificate [reject expired.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP wrong hostname in certificate [reject wrong.host.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP SHA-256 signature [accept sha256.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP 1000 subjectAltNames [accept 1000-sans.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP incomplete chain of trust [reject incomplete-chain.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP Superfish CA [reject superfish.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP eDellRoot CA [reject edellroot.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 SKIP DSDTestProvider CA [reject dsdtestprovider.badssl.com:443]
      reason: stub didn't reject a self-signed certificate
 PASS support for TLS server name indication (SNI) [accept tlsfun.de:443]
ERROR self-signed certificate [reject expired.tlsfun.de:443]
      reason: stub exited with return code 1
      output: Traceback (most recent call last):
                File "python2-urllib2/run.py", line 13, in <module>
                  urllib2.urlopen("https://" + host + ":" + port, cafile=cafile)
                File "/usr/lib64/python2.7/urllib2.py", line 154, in urlopen
                  return opener.open(url, data, timeout)
                File "/usr/lib64/python2.7/urllib2.py", line 437, in open
                  response = meth(req, response)
                File "/usr/lib64/python2.7/urllib2.py", line 550, in http_response
                  'http', request, response, code, msg, hdrs)
                File "/usr/lib64/python2.7/urllib2.py", line 475, in error
                  return self._call_chain(*args)
                File "/usr/lib64/python2.7/urllib2.py", line 409, in _call_chain
                  result = func(*args)
                File "/usr/lib64/python2.7/urllib2.py", line 558, in http_error_default
                  raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
              urllib2.HTTPError: HTTP Error 403: Forbidden
```

**Comment by jviide** [comment]
Seems to work for me now. Maybe this was a temporary error?

How we should react to such errors is a good question, though.

**Comment by ikisusi** [comment]
Still there. Did you try with something that actually accepts the certificate? :)

**Comment by jviide** [comment]
Ah, true, when testing a stub that **accepts** the `[reject expired.tlsfun.de:443]` certificate the error 403 does happen. My apologies :)

**Comment by Mamietti** [comment]
We'll comment it out for now. We'll get more info from the maintainer and add more tests before 0.4.0 release.

**Comment by aleksiklasila** [comment]
solved by #249
